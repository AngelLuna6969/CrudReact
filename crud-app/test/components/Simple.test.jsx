import { render } from '@testing-library/react';
import Simple from "../../src/components/Simple";

describe('Pruebas generales del componente Simple', () => {
    test('Se realiza comparacion con snapshot', () => {
        const { container } = render(<Simple titulo='Hello Word'
            subtitulo="Final de Cuatri" />);
        console.log(container);
        expect(container).toMatchSnapshot();
    });

    test('Se debe mostrar el titulo dentro de un h1', () => {
        const titulo = "Hola Mundo"
        const { container, getByText } = render(<Simple titulo={
            titulo} />)
        expect(getByText(titulo)).toBeTruthy();

        const h1 = container.querySelector("h1")
        expect(h1.innerHTML).toContain(titulo);
    });
});