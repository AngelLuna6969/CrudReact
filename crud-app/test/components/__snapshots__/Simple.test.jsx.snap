// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas generales del componente Simple Se realiza comparacion con snapshot 1`] = `
<div>
  <h1>
    Hello Word
  </h1>
  <h4>
    Final de Cuatri
  </h4>
  <p />
</div>
`;
